pm/sender.go

```diff
@@ -97,7 +100,7 @@ func (s *sender) CreateTicketBatch(sessionID string, size int) (*TicketBatch, er
 	}
 
 	if err := s.validateSender(); err != nil {
-		return nil, err
+		return nil, ErrSenderValidation
```

Comment:
We should make sure that the underlying error returned by `s.validateSender()` is not dropped here.

Instead of returning a specific error here, what if we returned a custom error type? Something like:

```
type SenderValidationError struct {
    error
}

...

func (s *sender) CreateTicketBatch(...) (...) {
    ...

    if err := s.validateSender(); err != nil {
        return nil, &SenderValidationError{err}
    }
}
```

If we use a custom error type, then the underlying error message can still be logged by the top level caller. The caller can also do a type assertion to check the error type.